Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm

```diff
@@ -1017,7 +1017,10 @@ - (void)makeCredentialWithClientDataHash:(NSData *)clientDataHash options:(_WKPu
     result.userVerificationString = toString(userVerification(options.userVerification));
     if (auto attachment = authenticatorAttachment(options.authenticatorAttachment))
         result.authenticatorAttachmentString = toString(*attachment);
-    result.extensions = authenticationExtensionsClientInputs(options.extensions);
+    if (options.extensionsCBOR)
+        result.extensions = WebCore::AuthenticationExtensionsClientInputs::fromCBOR(vectorFromNSData(options.extensionsCBOR));
```

Comment:
It is a bit unfortunate to copy the types from a NSData to a Vector, just to decode it.

Ideally, AuthenticationExtensionsClientInputs::fromCBOR() would take in a `Span<const uint8_t>` so we could pass in data without copying it. It is cheap to construct a Span from a Vector or a NSData because it doesn't involve copying.

Reply:
Ooh, that's really good to know. Updating CBORReader with this.

Reply:
We should check (as a follow-up patch) if there are other cases where we lack this optimization.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=242919